test/suites/test_crud.py

```diff
@@ -0,0 +1,584 @@
+                'args': ['tester', 'bucket_id', {'timeout': 10}],
+            },
+            'correct_output': {
+                'rows': [[4, 104, 'Ethan']],
+            },
+            'exception_input': {
+                'args': ['tester', 'no-idx', {'timeout': 10}],
+            },
+            'exception_output': {
+                'str': [
+                        r'BorderError: Index "no-idx" of space "tester" doesn\'t exist', 
+                    ],
+            },
+        },
+        'crud_len': {
+            'correct_input': {
+                'args': ['tester', {'timeout': 10}],
+            },
+            'correct_output': {
+                'scalar': 13,
+            },
+            'exception_input': {
+                'args': ['no-such-space-name', {'timeout': 10}],
+            },
+            'exception_output': {
+                'str': [
+                        r'LenError: Space "no-such-space-name" doesn\'t exist', 
+                    ],
+            },
+        },
+        'crud_count': {
+            'correct_input': {
+                'args': ['tester', [['==', 'bucket_id', 100]], {'timeout': 10}],
+            },
+            'correct_output': {
+                'scalar': 9,
+            },
+            'exception_input': {
+                'args': ['no-such-space-name', [['==', 'bucket_id', 100]], {'timeout': 10}],
+            },
+            'exception_output': {
+                'str': [
+                        r'CountError: Space "no-such-space-name" doesn\'t exist', 
+                    ],
+            },
+        },
+        'crud_unflatten_rows': {
+            'correct_input': {
+                'args': [
+                    [
+                        [1, 100, 'Mike'], 
+                        [2, 100, 'Mike'], 
+                        [3, 100, 'Mike'], 
+                        [4, 100, 'Mike'], 
+                        [5, 200, 'Bill'], 
+                        [6, 300, 'Rob'],
+                    ],
+                    [
+                        {'name': 'id', 'type': 'unsigned'}, 
+                        {'name': 'bucket_id', 'type': 'unsigned'}, 
+                        {'name': 'name', 'type': 'string'}
+                    ],
+                ],
+            },
+            'correct_output': {
+                'scalar': [
+                    {'bucket_id': 100, 'name': 'Mike', 'id': 1}, 
+                    {'bucket_id': 100, 'name': 'Mike', 'id': 2}, 
+                    {'bucket_id': 100, 'name': 'Mike', 'id': 3}, 
+                    {'bucket_id': 100, 'name': 'Mike', 'id': 4}, 
+                    {'bucket_id': 200, 'name': 'Bill', 'id': 5}, 
+                    {'bucket_id': 300, 'name': 'Rob', 'id': 6},
+                ],
+            },
+            'exception_input': {
+                'args': [[],[]],
+            },
+            'exception_output': {
+                'str': [],
+            },
+        },
+    }
+
+    def test_crud_module_via_connection(self):
```

Comment:
You should also run the same tests for connection_pool and mesh_connection (if the latter would be too time-consuming, you may omit it).

Reply:
Done, now implemented tests for `mesh connection` and `connection pool`